sleap/gui/commands.py

```diff
@@ -3470,13 +3481,230 @@ def do_action(context: CommandContext, params: dict):
         if rls is not None:
             context.openWebsite(rls.url)
 
+class ExportClipVideo(AppCommand):
+    @staticmethod
+    def do_action(context: CommandContext, params: dict):
+        """
+        Executes the export action for a video clip and its labels.
 
-def copy_to_clipboard(text: str):
-    """Copy a string to the system clipboard.
+        Args:
+            context (CommandContext): The application context containing the current state.
+            params (dict): Parameters for the export, including:
+                - 'filename' (str): The path to save the exported video.
+                - 'fps' (int): Frames per second for the exported video.
+                - 'open_when_done' (bool): Whether to open the video file after exporting.
+
+        Raises:
+            ValueError: If the frame range is invalid or no clip is selected.
+            RuntimeError: If there are issues with video writing or saving labels.
+        """
+        # Extract video and labels from context
+        video = context.state["video"]
+        labels = context.state["labels"]
 
-    Args:
-        text: String to copy to clipboard.
-    """
-    clipboard = QtWidgets.QApplication.clipboard()
-    clipboard.clear(mode=clipboard.Clipboard)
-    clipboard.setText(text, mode=clipboard.Clipboard)
+        # Ensure frame range is set; default to all frames if None
+        frame_range = context.state.get("frame_range", (0, video.frames))
+
+        # Validate frame range
+        if frame_range[0] < 0 or frame_range[1] > video.frames:
+            raise ValueError(f"Frame range {frame_range} is outside video bounds [0, {video.frames}]")
+
+        # Check if clip is selected, raise error if no clip selected
+        if frame_range == (0, video.frames) or frame_range == (0, 1) or frame_range[0] == frame_range[1]:
+            raise ValueError("No valid clip frame range selected! Please select a valid frame range using shift + click in the GUI.")
+
+        # Map frame indices to the actual labeled frame objects
+        frame_to_index = {lf.frame_idx: idx for idx, lf in enumerate(labels.labeled_frames) if lf.video == video}
+        valid_frame_indices = [frame for frame in range(*frame_range) if frame in frame_to_index]
+
+        if not valid_frame_indices:
+            raise ValueError("No valid labeled frames found in the selected frame range.")
+
+        # Extract only the selected frames into a new Labels object
+        pruned_labels = labels.extract(
+            inds=[frame_to_index[frame] for frame in valid_frame_indices],
+            copy=True  # Ensures a deep copy of the extracted labels
+        )
```

Comment:
```suggestion
        inds = (video, range(frame_range))

        # Extract only the selected frames into a new Labels object
        pruned_labels = labels.extract(
            inds=inds,
            copy=True  # Ensures a deep copy of the extracted labels
        )
```

since we want to run into this case in `Labels.extract` -> `Labels.__getitem__` -> `Labels.get`:
https://github.com/talmolab/sleap/blob/66d96ced50153e98b952176fbdb73043e105e397/sleap/io/dataset.py#L763-L764

which will call this case of `Labels.find` -> `Labels._cache.find_frames`:
https://github.com/talmolab/sleap/blob/66d96ced50153e98b952176fbdb73043e105e397/sleap/io/dataset.py#L142-L147